diff --git a/CMakeLists.txt b/CMakeLists.txt
index f088f6a..a8bcf80 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -156,10 +156,6 @@ add_library(CAF::internal ALIAS caf_internal)
 
 install(TARGETS caf_internal EXPORT CAFTargets)
 
-# -- create the libcaf_test target ahead of time for caf_core ------------------
-
-add_library(libcaf_test)
-
 # -- add uninstall target if it does not exist yet -----------------------------
 
 if(NOT TARGET uninstall)
@@ -277,8 +273,6 @@ function(caf_add_component name)
     target_compile_definitions(
       "caf-${name}-test" PRIVATE
       CAF_TEST_DATA_DIR="${CMAKE_CURRENT_SOURCE_DIR}/tests/data")
-    target_link_libraries("caf-${name}-test" PRIVATE libcaf_test
-                          ${CAF_ADD_COMPONENT_DEPENDENCIES} ${lib_target})
   endif()
   foreach(source_file ${CAF_ADD_COMPONENT_SOURCES})
     get_filename_component(ext ${source_file} EXT)
@@ -351,8 +345,6 @@ endfunction()
 
 add_subdirectory(libcaf_core)
 
-add_subdirectory(libcaf_test)
-
 if(CAF_ENABLE_NET_MODULE)
   add_subdirectory(libcaf_net)
 endif()
